prow.go

```diff
@@ -945,14 +1051,23 @@ func findTargetName(spec *corev1.PodSpec) (string, error) {
 				if len(value) > 0 {
 					value := (&resolvedEnvironment{env: container.Env}).Resolve(value)
 					if len(value) == 0 {
-						return "", fmt.Errorf("bug in resolving %s", value)
+						return "", false, fmt.Errorf("bug in resolving %s", value)
 					}
-					return value, nil
+					targetName = value
 				}
 			}
+			if strings.HasPrefix(arg, "--template") {
```

Comment:
This feels like a boolean that should be returned from this method or passed in.  Probably passed in.

Reply:
The result of this check is returned as a boolean by the method. To change to "passed in", are you suggesting a method signature change to something like:
```
func findTarget(spec *corev1.PodSpec, isTemplate *bool) (string, error)
     ...
```
and setting `*isTemplate` in the body?

Reply:
no, you should be telling findTarget what type of job this is.  You should know up front whether this is template or step, not guess.  Then you should either have two methods (if necessary) or one method.  But the end goal is not accidentally mistaking one job for another because someone changed something.  THe step vs template determination needs to be done one time at the top of the job, not later.